Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m

```diff
@@ -24,38 +24,30 @@
 #import "FIRMessaging.h"
 #import "FIRMessagingRemoteNotificationsProxy.h"
 
-#pragma mark - Expose Internal Methods for Testing
-// Expose some internal properties and methods here, in order to test
-@interface FIRMessagingRemoteNotificationsProxy ()
-
-@property(readonly, nonatomic) BOOL didSwizzleMethods;
-@property(readonly, nonatomic) BOOL didSwizzleAppDelegateMethods;
-
-@property(readonly, nonatomic) BOOL hasSwizzledUserNotificationDelegate;
-@property(readonly, nonatomic) BOOL isObservingUserNotificationDelegateChanges;
-
-@property(strong, readonly, nonatomic) id userNotificationCenter;
-@property(strong, readonly, nonatomic) id currentUserNotificationCenterDelegate;
-
-+ (instancetype)sharedProxy;
-
-- (BOOL)swizzleAppDelegateMethods:(id<UIApplicationDelegate>)appDelegate;
-- (void)listenForDelegateChangesInUserNotificationCenter:(id)notificationCenter;
-- (void)swizzleUserNotificationCenterDelegate:(id)delegate;
-- (void)unswizzleUserNotificationCenterDelegate:(id)delegate;
-
-void FCM_swizzle_appDidReceiveRemoteNotification(id self,
-                                                 SEL _cmd,
-                                                 UIApplication *app,
-                                                 NSDictionary *userInfo);
-void FCM_swizzle_appDidReceiveRemoteNotificationWithHandler(
-    id self, SEL _cmd, UIApplication *app, NSDictionary *userInfo,
-    void (^handler)(UIBackgroundFetchResult));
-void FCM_swizzle_willPresentNotificationWithHandler(
-    id self, SEL _cmd, id center, id notification, void (^handler)(NSUInteger));
-void FCM_swizzle_didReceiveNotificationResponseWithHandler(
-    id self, SEL _cmd, id center, id response, void (^handler)());
+#pragma mark - Invalid App Delegate or UNNotificationCenter
 
+@interface RandomObject : NSObject
```

Comment:
Not sure why we need to define a class for this, the purpose of passing a randomObject is to passing an invalid class, defining a serials of methods seem unnecessary.

Reply:
@chliangGoogle  To verify if the object methods were swizzled without using private details we need to invoke one of the methods and check its side effect.
E.g. on the [line 163](https://github.com/firebase/firebase-ios-sdk/blob/ad1da54d0fb34f72fb3139ef833a711fb8d555b1/Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m#L163) we perform the method and verify if [ [[FIRMessaging messaging] appDidReceiveMessage:userInfo] ](https://github.com/firebase/firebase-ios-sdk/blob/ad1da54d0fb34f72fb3139ef833a711fb8d555b1/Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m#L161) was invoked. If it was, that would mean that even the random object is swizzled.

Defining the methods on an object which does not conform to the required protocols looks like the simples way to implement it. Please let me know if you can see an easier way.